### PR TITLE
Add Kosh

### LINKED ISSUE #27: implement mahan kosh lookup
- [x] Add data to database
- [x] Add endpoint to API

### PR DESCRIPTION
**Pull request by ManjotS**
Fixes #27

### CODE REVIEW

File: api/controllers/kosh.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  const word = req.params.Word;
+  try {
+    conn = await req.app.locals.pool.getConnection();
+    const q = `SELECT w.ID AS id, w.Word AS word, w.WordUni AS wordUni,
+                  d.DefGurmukhi AS definition, d.DefGurmukhiUni AS definitionUni
+                FROM MahanKoshWords w
+                LEFT JOIN MahanKoshDefinitions d ON w.Definition = d.ID
+                WHERE w.Word LIKE '${word}%' OR w.WordUni LIKE BINARY '${word}%'
+                ORDER BY w.ID`;
+    const rows = await conn.query(q, []);
```
Comment (tsingh777) [inline api/controllers/kosh.js:40]: shouldn't the `word` from the request be used in the second param for query? if not you aren't sanitizing the input.

File: api/controllers/kosh.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                  d.DefGurmukhi AS definition, d.DefGurmukhiUni AS definitionUni
+                FROM MahanKoshWords w
+                LEFT JOIN MahanKoshDefinitions d ON w.Definition = d.ID
+                WHERE
+                  w.Word LIKE '%${query}%' OR
+                  w.WordUni LIKE BINARY '%${query}%' OR
+                  d.DefGurmukhi LIKE '%${query}%' OR
+                  d.DefGurmukhiUni LIKE BINARY '%${query}%'
+                ORDER BY w.ID`;
+    const rows = await conn.query(q, []);
```
Comment (tsingh777) [inline api/controllers/kosh.js:64]: pass param to prepared statement

File: api/controllers/kosh.js (hunk 1 above)
Comment (tsingh777) [reply api/controllers/kosh.js:40]: pass param to prepared statement

File: api/controllers/kosh.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      false,
+    );
+  }
+  try {
+    conn = await req.app.locals.pool.getConnection();
+    const q = `SELECT ID AS id, Word AS word, WordUni AS wordUni
+                FROM MahanKoshWords
+                WHERE FirstLetter='${letter}' OR FirstLetterUni='${letter}'
+                ORDER BY ID`;
+    const rows = await conn.query(q, []);
```
Comment (tsingh777) [inline api/controllers/kosh.js:20]: pass param to prepared statement

File: api/controllers/kosh.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
+const lib = require('../lib');
+
+exports.letter = async (req, res) => {
+  let conn;
+  const letter = req.params.Letter;
+  if (letter.length !== 1) {
+    lib.error(
+      'You must supply a single character to see the index for that letter.',
+      res,
+      413,
```
Comment (tsingh777) [inline api/controllers/kosh.js:10]: I'm not sure if this is what 413 is really for, I think it's meant for size in terms of bytes. https://tools.ietf.org/html/rfc7231#section-6.5.11 `413 Payload Too Large` 

Would 422 `422 Unprocessable Entity` be be more appropriate? https://tools.ietf.org/html/rfc4918#section-11.2

Would like to hear other's opinions.

Comment (ManjotS) [reply api/controllers/kosh.js:10]: i debated this a few times..

**Comment by ManjotS** [comment]
i described the design in #27

File: api/controllers/kosh.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  } catch (err) {
+    lib.error(err, res, 500);
+  } finally {
+    if (conn) conn.end();
+  }
+};
+
+exports.word = async (req, res) => {
+  let conn;
+  const word = `${req.params.Word}%`;
```
Comment (tsingh777) [inline api/controllers/kosh.js:31]: potential we should sanitize this a bit more. Can follow up later though.

File: api/controllers/kosh.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  } catch (err) {
+    lib.error(err, res, 500);
+  } finally {
+    if (conn) conn.end();
+  }
+};
+
+exports.search = async (req, res) => {
+  let conn;
+  const query = `%${req.params.query}%`;
```
Comment (tsingh777) [inline api/controllers/kosh.js:51]: potential we should sanitize this a bit more. Can follow up later though.

**Pull request merged**